migrator/src/main/java/com/google/phonenumbers/migrator/MigrationFactory.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2020 The Libphonenumber Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.phonenumbers.migrator;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.i18n.phonenumbers.metadata.RangeSpecification;
+import com.google.i18n.phonenumbers.metadata.i18n.PhoneRegion;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * Factory class to instantiate {@link MigrationJob} objects. To create a Migration Job, a two digit
+ * BCP-47 region code is required as well as a number input (either a single E.164 number or a file
+ * path containing one E.164 number per line). There is also the option for specifying a custom
+ * recipes file as a third parameter to use for migrations instead of the default recipes file.
+ */
+public class MigrationFactory {
```

Comment:
Let's chat about the value of static factory classes in a VC. There's a lot to talk about here.